src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApi.XmlModel.cs

```diff
@@ -10,12 +10,14 @@ public partial class JavaApi
 	{
 		public JavaApi ()
 		{
-			Packages = new List<JavaPackage> ();
+			Packages = new Dictionary<string, JavaPackage> (StringComparer.OrdinalIgnoreCase);
 		}
 
 		public  string?                 ExtendedApiSource   { get; set; }
 		public  string?                 Platform            { get; set; }
-		public  IList<JavaPackage>      Packages            { get; set; }
+		public  IDictionary<string, JavaPackage>      Packages            { get; }
```

Comment:
Should this mirror `JavaPackage`, and have `JavaApi.Packages` be an `IReadOnlyDictionary<string, JavaPackage>` with a corresponding `AddPackage(JavaPackaage)` method?

It may also make for a "smaller diff" if `Packages` remains `IEnumerable<JavaPackage>`-compatible, and make `AllPackages` the `Dictionary<string, JavaPackage>`-ish member.

Reply:
Related? it may be interesting to consider doing a [`KeyedCollection<TKey, TItem>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.keyedcollection-2?view=net-5.0) subclass:

```csharp
internal class JavaPackages : KeyedCollection<string, JavaPackage> {
    protected override string GetKeyForItem (JavaPackage package) => package.Name;
}
```

Reply:
A concern here -- seemingly minor -- is that this assembly is used in xamarin-android: https://github.com/xamarin/xamarin-android/blob/97190b6cc56ef81be82bab48737155089a16f500/build-tools/api-xml-adjuster/Program.cs

Fortunately `Program.cs` doesn't appear to use these properties, but it is a possible concern for cross-repo compatibility.

Reply:
There is a XA run with this change in the PR description.  It built, so I'm assuming it works.  😄